### DISCUSSION TITLE
Spanish_Spain partial translation!

### DISCUSSION THREAD

**Discussion by El-Juancho** [Marat-Tanalin/bsnes-mt#13] [Show and tell]
Hello, I have done a partial translation in the bsnes-mt emulator (Spanish_Spain)

I'm writing this here because I have no idea how to do a pull request on github.

This is the link of the translation I have done, greetings.

https://drive.google.com/file/d/1IQJu4V8WgqqNDr0JbnkneLG2NiCwHqwA/view?usp=sharing

**Comment by Marat-Tanalin** [comment]
Thanks, @JuanchoES. Looks fine in general. One thing I’m not sure about: the translation of the “area” word in the “Show Overscan Area” phrase seems to be _plural_ (at least that’s what I can see when translating your translation “Mostrar las areas de sobreescaneo” back to English and to Russian) in your translation while being (and meant to be) _singular_ in the canonical English set and translations. Is that plural form intended?

**Reply by El-Juancho** [reply]
I thought it would be more correct in plural, I don't understand much about that option but when I activate that option 2 overscan areas appear? One above and one below, in singular it would be "Mostrar area de sobreescaneo"








![Sin título](https://user-images.githubusercontent.com/70743149/103442188-4693fe80-4c54-11eb-8345-92d7fa8ca9a6.png)

**Reply by Marat-Tanalin** [reply]
Thanks. It’s important for a translation to correctly represent the original phrase, just in another language. Also, Afaik, in general, overscan may be both vertical and horizontal, being actually a _contiguous_ extra area around the all four image edges, not just above and below. So let’s keep overscan area singular as it is in the canonical English variant.

**Comment by Marat-Tanalin** [comment]
[Merged](https://github.com/Marat-Tanalin/bsnes-mt/commit/8f0e17fdd53dc31ce279794805201a49ebdcc0f2), thanks.

**Comment by El-Juancho** [comment]
Hello again! I have done a 99% translation!!! it took more time than i expected haha.

https://drive.google.com/file/d/1hKMXZQYdkcaKBe_OwF0iKYOWib5OXpAu/view?usp=sharing

-The "Show overscan area" is now singular.
-Some tooltips are not translated because it would take me much longer to do the translation.
-Some words in english are untranslatable because the word in spanish does not exist, like skew (Settings.Audio.Skew) 
in the translation I have put "static speed control".
-Some words in english don't need to be translated because in spanish they sound weird, we use some english words in our vocabulary.

Please, if you see something that you don't like tell me, i will fix it :)

**Reply by Marat-Tanalin** [reply]
Thanks. Your new fuller translation looks fine in general, but there are some things that need clarification or fixing (please apply changes to **[the&nbsp;file](https://github.com/Marat-Tanalin/bsnes-mt/files/5767938/es.txt)** attached to this comment since it already has some fixes compared with your original one):

* `Settings.Common.AssignLowercase = "Asignar"` — this string is meant to be lowercase, so it should probably be "asignar" (with lowercase leading "a") instead of "Asignar".
* `Settings.Audio.Skew` — "Control de velocidad estática" is somewhat too long. I think you could treat the original phrase as "Sample rate" or just "Rate".
* `Settings.Input.TurboRate = "velocidad del Turbo"` — looks like there is something wrong with capitalization. Somehow the first letter of the first word is lowercase and the first letter of the last word is uppercase. This is probably not intended and should be "Velocidad del turbo" — with the same capitalization as in the original English string.
* `Settings.Emulator.Rewind.Frequency.everyFrames = "todos | fotogramas"` — the first letter of the first word is uppercase in the original English string, so it should probably be uppercase in Spanish too.
* `Settings.Emulator.Rewind.Length.states = "| Estado"` — the first letter of the word is lowercase in the original English string, so it should probably be lowercase in Spanish too ("|" is automatically replaced with a number when the string is displayed, so the word in the translation string is **not** first in the sentence).
* `Settings.Compatibility.cpu.FastMath  = "matemática rápida"` — the first letter of the first word is uppercase in the original English string, so it should probably be uppercase in Spanish too.
* `Settings.Drivers.changeConfirm` — machine translation of the "quites" word to English and Russian results in a term that has the main meaning of "delete"/"erase". There is probably a translation that would mean "unload"/"close" in a less ambiguous way.
* `Settings.Drivers.syncModePresets.AdaptiveSync = "Adaptive sync"` — is different from the original English string just by the case of the first letter of the "sync" word. Not sure what this means: either the Spanish translation is identical to the original English string (as you said in your previous comment) and you intentionally lowercased the "sync" word, or this is not intended and the string is actually not translated.
* `Tools.LoadState.RemoveAllStates` and `Tools.LoadState.RemoveAllStates.confirm.title` translations contain the same string (except for the question mark), though in the original English strings, the first string is "Remove **All** States" and the second one is "Remove **quick** states?". The `Tools.LoadState.RemoveAllStates.confirm.title` string should contain a translation of "quick", not "all".
* `Tools.CheatEditor.EditCheat`, `Tools.CheatEditor.Edit` — contains "Editar" as a translation of the original English "Edit", but machine translation translates "Editar" as a term that means "publish". There is a probability that your translation does not take the context into account, or "Editar" contains a typo. For more clarity: "Edit" means "Change" or "Modify" here.
* `Browser.OpenSnesRom  = "Abir ROM de SNES"` — machine translation does not understand the "Abir" word. So the word probably contains a typo.
* `Rewind.HistoryExhausted`— looks like not translated (identical to the corresponding English string). Is this meant to be not translated, or is the Spanish translation meant to be identical to the English string?

I already applied some changes:

* Some strings did contain wrong quote characters (code-style `""` instead of typography-correct `“”` used in the original English strings) that did conflicted with the translation-file syntax where quote characters are used as **delimiters** wrapping the string value, so such quote characters **inside** the string itself did break parsing the translation after the first quote character inside the string. I replaced the wrong quote characters back to typography-correct ones as they were in the original English strings.
* Some strings contained unneeded leading space characters, I removed them.
* I added the underscore (`_`) prefix to keys of untranslated strings (mainly tooltips as you said in your previous comment). Generally, translations should not contain untranslated strings at all because this leads to unneeded duplication (the canonical English translation file already has those strings) and also to confusion in terms of whether the strings are untranslated or the legit translation is just identical to the original English string. But for making translating them later easier with no need for copy-pasting from `en.txt` with maintaining their position in the file, I decided to keep them in this case (given that the number of untranslated strings is low) and only added the prefix so that it’s immediately clear that the strings are untranslated (I would probably prefer to comment them out, but my translation-file syntax and parser don’t support comments yet).

**Reply by El-Juancho** [reply]
> * `Settings.Audio.Skew` — "Control de velocidad estática" is somewhat too long. I think you could treat the original phrase as "Sample rate" or just "Rate".

I change it to "Frecuencia de muestreo" if it is still too long i recommend putting just "Frecuencia".
> * `Tools.CheatEditor.EditCheat`, `Tools.CheatEditor.Edit` — contains "Editar" as a translation of the original English "Edit", but machine translation translates "Editar" as a term that means "publish". There is a probability that your translation does not take the context into account, or "Editar" contains a typo. For more clarity: "Edit" means "Change" or "Modify" here.

"Editar trucos" means change/edit/modify cheat codes, it is correct.

All of the other things that you posted are fixed. In the next version I want to translate all the tooltips, later in a different version I will make a revision of the text.



https://drive.google.com/file/d/1Cb2YJ2LP18mC2kWiNo-NKAFZ4qzcZ4tB/view?usp=sharing

**Reply by Marat-Tanalin** [reply]
Thanks. For a couple of strings, it’s still unclear whether the string is untranslated or it’s a legit translation just _looking_ identical to the original English string:

* `Settings.Drivers.syncModePresets.AdaptiveSync`
* `Rewind.HistoryExhausted`

**Reply by El-Juancho** [reply]
Adaptative Sync is correct.

The "Rewind.HistoryExhausted" string is translated but get ride of the "_".
![image](https://user-images.githubusercontent.com/70743149/103826896-0e3b5880-5078-11eb-86cd-926226c2a41d.png)

**Reply by Marat-Tanalin** [reply]
[Merged](https://github.com/Marat-Tanalin/bsnes-mt/commit/e6f1c2c6329280a35752416b333efd612f913208). Thank you, @JuanchoES.

**Comment by Marat-Tanalin** [comment]
Included in the [1.3.3r2](https://github.com/Marat-Tanalin/bsnes-mt/releases/tag/1.3.3) release.

**Comment by El-Juancho** [comment]
A revision of the text + adding my real name, the next update should be the last.

https://drive.google.com/file/d/1gXMXg_Y_r_7OaNNcHiurhO87NbVJP5JZ/view?usp=sharing

**Comment by Marat-Tanalin** [comment]
> the next update should be the last.

Could you provide some ETA for your planned next/last update? I planned to merge them all at once. 🙂

**Reply by El-Juancho** [reply]
on the weekend
